project-koku/koku
# [COST-2620] update middleware to handle org_id

Fixes [COST-2620](https://issues.redhat.com/browse/COST-2620)

Changes proposed in this PR:
* Get org_id from auth header
* Include org_id in create_customer
* change cache key to a tuple of `(account, org_id)`

Testing Instructions:
1. Bring up koku and related services, `make docker-up-min-trino`
2. Connect to the db and check your results in the `api_customer` table: 
     - `SELECT * FROM api_customer;`
     - if you have some data, you may see an entry similar to this: 
```
[ RECORD 1 ]+-------------------------------------
id           | 1
date_created | 2022-06-08 12:55:49.994707+00
date_updated | 2022-06-08 12:55:49.994871+00
uuid         | c8a69de3-da9b-4759-b7de-f2e5faa4b0dc
account_id   | 10001
schema_name  | acct10001
org_id       | 
```

4. Using Postman (or any other way to send requests), send a few GET requests to any endpoint with a custom `x-rh-identity` header using base64 encoded identity headers. A few examples:
     - Example endpoint: `localhost:8000/api/cost-management/v1/reports/aws/costs/`
     - An identity without an org_id: ```'{"identity": {"account_number": "10005", "type": "User", "user": {"username": "user_dev", "email": "user_dev@foo.com", "is_org_admin": "True", "access": {}}},"entitlements": {"cost_management": {"is_entitled": "True"}}}'```
     - An identity with an org_id: ```'{"identity": {"account_number": "10006", "org_id": "1234567", "type": "User", "user": {"username": "user_dev", "email": "user_dev@foo.com", "is_org_admin": "True", "access": {}}},"entitlements": {"cost_management": {"is_entitled": "True"}}}'```
5. After sending the requests with the headers, check the `api_customer` table in the db again and you should see entries for the different headers sent.
6. `SELECT * FROM api_customer;`
```
-[ RECORD 4 ]+-------------------------------------
id           | 4
date_created | 2022-06-08 12:59:13.421117+00
date_updated | 2022-06-08 12:59:13.421182+00
uuid         | d0a01380-c13f-4771-80e5-d199d7c26ccf
account_id   | 10005
schema_name  | acct10005
org_id       | 
-[ RECORD 5 ]+-------------------------------------
id           | 5
date_created | 2022-06-08 13:05:57.725407+00
date_updated | 2022-06-08 13:05:57.725441+00
uuid         | a9036f04-6f05-4ce5-8cde-642665ffcccf
account_id   | 10006
schema_name  | acct10006
org_id       | 1234567
```

---
Additional Context:
A followup will be necessary after some DB changes to only base things off org_id instead of account_id

Smokes run from before label removed: https://ci.ext.devshift.net/job/project-koku-koku-pr-check/5516/
